src/RemoteRequest/TemporaryFileCachedRemoteGetRequest.php

```diff
@@ -0,0 +1,201 @@
+     * @var string
+     */
+    private $directory;
+
+    /**
+     * The temporary file abspath.
+     *
+     * @var string
+     */
+    private $file;
+
+    /**
+     * Instantiate a TemporaryFileCachedRemoteGetRequest object.
+     *
+     * @param RemoteGetRequest $remoteGetRequest The decorated RemoteGetRequest object.
+     * @param string           $directory        Optional. The directory abspath where the temp file would be saved.
+     */
+    public function __construct(RemoteGetRequest $remoteGetRequest, $directory = '')
+    {
+        $this->remoteGetRequest = $remoteGetRequest;
+        $this->directory        = $directory ? $directory : sys_get_temp_dir();
+    }
+
+    /**
+     * Get the cached request from a temporary file.
+     *
+     * @param string $url     URL to get.
+     * @param array  $headers Optional. Associative array of headers to send with the request. Defaults to empty array.
+     * @return Response Response for the executed request.
+     * @throws FailedToGetCachedResponse If retrieving the contents from the cache failed.
+     */
+    public function get($url, $headers = [])
+    {
+        $filename = self::CACHED_FILE_PREFIX . md5($url . json_encode($headers));
+
+        $this->file = $this->directory . DIRECTORY_SEPARATOR . $filename;
+
+        if (! file_exists($this->file)) {
+            return $this->getRemoteResponse($url, $headers);
+        }
+
+        $cachedResponse = file_get_contents($this->file);
+
+        if (false !== $cachedResponse) {
+            if (PHP_MAJOR_VERSION >= 7) {
+                $cachedResponse = unserialize($cachedResponse, [ RemoteGetRequestResponse::class ]);
+            } else {
+                // PHP 5.6 does not provide the second $options argument yet.
+                $cachedResponse = unserialize($cachedResponse);
+            }
+        }
+
+        if (! $cachedResponse instanceof RemoteGetRequestResponse || $this->isExpired($cachedResponse)) {
+            return $this->getRemoteResponse($url, $headers);
+        }
+
+        return $cachedResponse;
+    }
+
+    /**
+     * Get the remote response using the decorated RemoteGetRequest object.
+     *
+     * @param string $url     URL to get.
+     * @param array  $headers Associative array of headers to send with the request.
+     * @return Response Response for the executed request.
+     */
+    private function getRemoteResponse($url, $headers)
+    {
+        try {
+            $response = $this->remoteGetRequest->get($url, $headers);
+            $this->cacheResponse($response);
+            return $response;
+        } catch (Exception $error) {
+            throw FailedToGetCachedResponse::withUrl($url);
+        }
+    }
+
+    /**
+     * Save the response in temporary file.
+     *
+     * @param Response $response The response that need to be cached.
+     */
+    private function cacheResponse(Response $response)
+    {
+        file_put_contents($this->file, serialize($response));
```

Comment:
As discussed above, `$this->file` should not be a class property, but rather passed in as an argument, as it is not consistent for the lifetime of the object.

Alternatively, this could directly make use of the `getTemporaryFilePath($url)` method I suggested above.